### PR TITLE
feat(websocket): MED-174 replace websockets with api call in favorite header

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
- refacto home controller
- remove unused calls + websockets
- add calls to favorite service

## Checklist tests
use favorite header as usual

## Issue ticket number and link
[MED-174](https://jira.support-ent.fr/browse/MED-174)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/home.ts [hunk 1]
```diff
@@ -17,120 +22,187 @@ interface ViewModel {
...
+            ])
+            Utils.safeApply(this.$scope);
+        } catch (e) {
+
+        }
+        //
+        //
+        // this.$scope.ws.send(new Frame('textbooks', 'get', [], {}));
+        // //     $scope.ws.send(new Frame('favorites', 'get', [], {}));
+        // this.$scope.ws.send(new Frame('search', 'PLAIN_TEXT', ['fr.openent.mediacentre.source.GAR'], {"query": ".*"}));
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/home.ts:95]: other websockets will be removed in next tickets

File: src/main/resources/public/ts/controllers/home.ts [hunk 2]
```diff
@@ -17,120 +22,187 @@ interface ViewModel {
...
+            //         this.$scope.hc.mc.favorites = []
+            //     } else {
+            //         this.$scope.hc.mc.favorites = frame.data;
+            //         this.$scope.hc.mc.favorites.map((favorite) => {
+            //             favorite.favorite = true;
+            //         });
+            //     }
+            //     Utils.safeApply(this.$scope);
+            // }
+            search_Result = (frame) => {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/home.ts:152]: other websockets will be removed in next tickets

File: src/main/resources/public/ts/controllers/home.ts [hunk 3]
```diff
@@ -17,120 +22,187 @@ interface ViewModel {
...
+            this.textbooks = [];
+            Utils.safeApply(this.$scope);
+            // $scope.ws.send(new Frame('textbooks', 'refresh', [], {}));
         };
 
-        vm.seeMyExternalResource = (): void => {
-            $scope.ws.send(new Frame('search', 'PLAIN_TEXT', ['fr.openent.mediacentre.source.GAR'], {"query": ".*"}));
-            $location.path(`/search/plain_text`);
+        seeMyExternalResource = (): void => {
+            // this.$scope.ws.send(new Frame('search', 'PLAIN_TEXT', ['fr.openent.mediacentre.source.GAR'], {"query": ".*"}));
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/home.ts:176]: will be removed in next tickets

File: src/main/resources/public/ts/controllers/main.ts [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
 declare const window: any;
+declare var mediacentreUpdateFrequency: number;
 
-export interface Scope extends IRootScopeService {
+export interface MainScope extends IRootScopeService {
 	isStatusXXX(response: any, status: number): any;
 	getDataIf200(response: any): any;
-    hasShareRightView(signet: Signet[]): boolean;
-    hasShareRightManager(signet: Signet[]): boolean;
+	hasShareRightView(signet: Signet[]): boolean;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/main.ts:16]: indentation

File: src/main/resources/public/ts/controllers/main.ts [hunk 5]
```diff
@@ -1,27 +1,29 @@
...
-    removeLevelFromCourse(level: Label): void;
+	removeLevelFromCourse(level: Label): void;
 	removeDisciplineFromCourse(discipline: Label): void;
 	removeWordFromCourse(word: Label): void;
-    query: any;
-    addKeyWord(event: any): void;
-    disciplines: Labels;
-    levels: Labels;
-    displayDate (dateToFormat: Date): string;
+	query: any;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/main.ts:22]: indentation

File: src/main/resources/public/ts/controllers/main.ts [hunk 6]
```diff
@@ -155,108 +219,82 @@ export const mainController = ng.controller('MainController', ['$scope', 'route'
...
+			this.signet.plain_text = _.without(this.signet.plain_text, word);
+			if(this.signet.plain_text.length == 0) {
+				this.signet.plain_text = new Labels();
+				this.signet.plain_text.all = [];
 			}
 		};
 
 		// Utils
 
-		$scope.safeApply = function () {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/main.ts:247]: use the one from Utils instead

File: src/main/resources/public/ts/controllers/search.ts [hunk 7]
```diff
@@ -113,18 +114,18 @@ export const searchController = ng.controller('SearchController', ['$scope', '$l
...
-        $scope.ws.onmessage = (message) => {
-            const {event, state, data, status, error} = JSON.parse(message.data);
-            if ("ok" !== status) {
-                vm.loaders[error.source] = false;
-                $scope.safeApply();
-                throw JSON.parse(message.data).error;
-            }
-            if (event in eventResponses) eventResponses[event](new Frame(event, state, [], data));
-        };
+        // $scope.ws.onmessage = (message) => {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/search.ts:120]: will be removed in [next ticket]([MED-175](https://jira.support-ent.fr/browse/MED-175))

File: src/main/resources/public/ts/model/Socket.ts [hunk 8]
```diff
@@ -51,12 +51,12 @@ export class Socket {
                 this.callbacks.onmessage(message);
             }
         };
-        this._ws.onopen = (message: Event) => {
-            this.connected = true;
-            if ('onopen' in this.callbacks && this.callbacks.onopen !== undefined) {
-                this.callbacks.onopen(message);
-            }
-        };
+        // this._ws.onopen = (message: Event) => {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/model/Socket.ts:54]: will be removed in next [ticket]([MED-180](https://jira.support-ent.fr/browse/MED-180))

**Review by simonchaval — APPROVED** [review]

File: src/main/resources/public/ts/controllers/home.ts [hunk 9]
```diff
@@ -17,120 +22,157 @@ interface ViewModel {
...
+        this.signetLimit = this.mobile ? 4 : 8;
+
+    }
+
+    async $onInit() {
+        this.updateFrequency = mediacentreUpdateFrequency;
+
+        try {
+            await Promise.all([
+                await this.syncResources()
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/home.ts:81]: will be joined by other calls later

File: src/main/resources/public/ts/controllers/home.ts [hunk 10]
```diff
@@ -17,120 +22,157 @@ interface ViewModel {
...
 
-        $scope.$on('addFavorite', function (event, resource) {
-            $scope.mc.favorites.push(resource);
+        this.$scope.$on('addFavorite', async (event, resource) => {
+            await this.syncResources();
         });
 
+        this.$interval(async (): Promise<void> => {
+            await this.syncResources();
+            console.log("home $interval");
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/home.ts:113]: Commentaire

File: src/main/resources/public/ts/controllers/home.ts [hunk 11]
```diff
@@ -17,120 +22,157 @@ interface ViewModel {
...
+    async syncResources() {
+        try {
+            this.mainScope.mc.favorites = await this.favoriteService.get();
+            this.mainScope.mc.favorites.forEach((resource: Resource) => {
+                resource.favorite = true;
+            });
+            Utils.safeApply(this.$scope);
+        } catch (e) {
+            console.error("An error has occurred during fetching favorite ", e);
+            toasts.warning(lang.translate("mediacentre.error.favorite.retrieval")); //todo
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/home.ts:128]: Todo ?

File: src/main/resources/public/ts/controllers/home.ts [hunk 12]
```diff
@@ -17,120 +22,157 @@ interface ViewModel {
...
-                vm.orientationSignets = vm.orientationSignets.concat(vm.sharedSignets.filter(el => el.document_types[0] === "Orientation"));
-                $scope.safeApply();
+            search_Result = (frame) => {
+                this.displayedResources = frame.data.resources;
+                Utils.safeApply(this.$scope);
+            }
+            signets_Result = async (frame) => {
+                this.signets.all = this.publicSignets = this.orientationSignets = this.sharedSignets = [];
+                await this.signets.sync();
+                this.signets.all = this.signets.all.filter(signet => !signet.archived && signet.collab && signet.owner_id != model.me.userId);
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/home.ts:143]: typage de signet

File: src/main/resources/public/ts/controllers/home.ts [hunk 13]
```diff
@@ -17,120 +22,157 @@ interface ViewModel {
...
+                this.signets.formatSharedSignets(this.sharedSignets);
+                this.publicSignets = frame.data.signets.resources.filter(el => el.document_types[0] === "Signet");
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/home.ts:145]: typage el

File: src/main/resources/public/ts/directives/ResourceCard.ts [hunk 14]
```diff
@@ -157,7 +157,7 @@ export const ResourceCard = ng.directive('resourceCard',
                     $scope.ngModel.source == "fr.openent.mediacentre.source.Signet" ? parseInt($scope.ngModel.id) : $scope.ngModel.id;
                     let response = await FavoriteService.delete($scope.ngModel.id, $scope.ngModel.source);
                     if (response.status === 200) {
-                        $scope.ngModel.favorite = false;
+                        // $scope.ngModel.favorite = false;
```
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/directives/ResourceCard.ts:160]: tu le gardes com?

File: src/main/resources/public/ts/directives/ResourceCard.ts [hunk 15]
```diff
@@ -146,7 +146,7 @@ export const ResourceCard = ng.directive('resourceCard',
                     $scope.ngModel.source == "fr.openent.mediacentre.source.Signet" ? parseInt($scope.ngModel.id) : $scope.ngModel.id;
                     let response = await FavoriteService.create($scope.ngModel, $scope.ngModel.id);
                     if (response.status === 200) {
-                        $scope.ngModel.favorite = true;
+                        // $scope.ngModel.favorite = true;
```
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/directives/ResourceCard.ts:149]: tu le gardes com?

File: src/main/resources/public/ts/controllers/home.ts (hunk 10 above)
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/controllers/home.ts:113]: normal de garder ce console log?

File: src/main/resources/public/ts/controllers/home.ts (hunk 11 above)
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/controllers/home.ts:128]: pas besoin d'un lang.translate dans un toasts de memoire

File: src/main/resources/public/ts/directives/ResourceCard.ts (hunk 15 above)
Comment (BastienBenard) [inline src/main/resources/public/ts/directives/ResourceCard.ts:149]: si c'est pas utile, tu peux delete.

File: src/main/resources/public/ts/controllers/home.ts (hunk 12 above)
Comment (alicedraillard) [reply src/main/resources/public/ts/controllers/home.ts:143]: sera traité dans le ticket dédié aux signets

File: src/main/resources/public/ts/controllers/home.ts [hunk 16]
```diff
@@ -17,120 +22,156 @@ interface ViewModel {
...
+        this.signetLimit = this.mobile ? 4 : 8;
+
+    }
+
+    async $onInit() {
+        this.updateFrequency = mediacentreUpdateFrequency;
+
+        try {
+            await Promise.all([
+                await this.syncResources()
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/home.ts]: Tu n'as pas besoin de mettre un await, le promise.all va le faire.

**Review by BastienBenard — APPROVED** [review]

**Review by Sofianebernoussi — APPROVED** [review]

**Pull request merged**